### PR TITLE
Staff now have access to donor items

### PR DESCRIPTION
**Pull request by Crossedfall**
## About The Pull Request

Time is money and staff donate a lot of their time. It makes sense for staff to be granted access to donor specific items.

## Why It's Good For The Game

Adds another small incentive to being part of the mainline team

## Changelog
:cl:
add: Being on staff grants you access to donor items
/:cl:

### CODE REVIEW

**Comment by aidan-bowman** [comment]
time to become a mentor for free hats

**Comment by ike709** [comment]
I support this for admins but idk about mentors.

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -2021,7 +2021,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	var/datum/loadout_category/DLC = GLOB.loadout_categories["Donator"] // stands for donator loadout category but the other def for DLC works too xD
 	if(!LAZYLEN(GLOB.patrons) || !CONFIG_GET(flag/donator_items)) // donator items are only accesibile by servers with a patreon
```
Comment (Archanial) [inline code/modules/client/preferences.dm:2022]: since staff would have access to it, we shouldn't check if patrons is empty

**Comment by Archanial** [comment]
> 
> 
> I support this for admins but idk about mentors.

My fear is that people would apply for mentors just to get donator items since being a mentor doesn't really require anything from you and it is quite easy to become one.

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (Crossedfall) [reply code/modules/client/preferences.dm:2022]: Nah

**Comment by Crossedfall** [comment]
> > I support this for admins but idk about mentors.
> 
> My fear is that people would apply for mentors just to get donator items since being a mentor doesn't really require anything from you and it is quite easy to become one.

Time to up the requirements and expectations then.

**Comment by aidan-bowman** [comment]
> > > I support this for admins but idk about mentors.
> > 
> > 
> > My fear is that people would apply for mentors just to get donator items since being a mentor doesn't really require anything from you and it is quite easy to become one.
> 
> Time to up the requirements and expectations then.

no? mentoring is literally un-abusable other than this. what are you gonna do, ban someone? there is literally nothing negative a mentor can do, only positive, and that's why a higher bar can only be a bad thing

**Comment by OceanFish1** [comment]
> > > > I support this for admins but idk about mentors.
> > > 
> > > 
> > > My fear is that people would apply for mentors just to get donator items since being a mentor doesn't really require anything from you and it is quite easy to become one.
> > 
> > 
> > Time to up the requirements and expectations then.
> 
> no? mentoring is literally un-abusable other than this. what are you gonna do, ban someone? there is literally nothing negative a mentor can do, only positive, and that's why a higher bar can only be a bad thing

No donor items for mentors allowed
literally 1984

**Comment by aidan-bowman** [comment]
> > > > > I support this for admins but idk about mentors.
> > > > 
> > > > 
> > > > My fear is that people would apply for mentors just to get donator items since being a mentor doesn't really require anything from you and it is quite easy to become one.
> > > 
> > > 
> > > Time to up the requirements and expectations then.
> > 
> > 
> > no? mentoring is literally un-abusable other than this. what are you gonna do, ban someone? there is literally nothing negative a mentor can do, only positive, and that's why a higher bar can only be a bad thing
> 
> No donor items for mentors allowed
> literally 1984

that wasn't the point you monke, my man was talkin about making mentor harder to get

**Comment by TeomanTheGreat** [comment]
> > > I support this for admins but idk about mentors.
> > 
> > My fear is that people would apply for mentors just to get donator items since being a mentor doesn't really require anything from you and it is quite easy to become one.
> 
> Time to up the requirements and expectations then.

Mentor purge when

**Comment by Sinestia** [comment]
> 
> 
> > > > > > I support this for admins but idk about mentors.
> > > > > 
> > > > > 
> > > > > My fear is that people would apply for mentors just to get donator items since being a mentor doesn't really require anything from you and it is quite easy to become one.
> > > > 
> > > > 
> > > > Time to up the requirements and expectations then.
> > > 
> > > 
> > > no? mentoring is literally un-abusable other than this. what are you gonna do, ban someone? there is literally nothing negative a mentor can do, only positive, and that's why a higher bar can only be a bad thing
> > 
> > 
> > No donor items for mentors allowed
> > literally 1984
> 
> that wasn't the point you monke, my man was talkin about making mentor harder to get

In theory mentors are staff and representative of the server, that actually does mean a high standard is warranted. I'm sure like half the opposition to giving mentors donor stuff is because mentors have a poor reputation. 

It's not like mentors are even needed to get or give advice, you have OOC on golden and on sage it's preferable to ask other players if possible since it'll result in actual interaction (Plus LOOC.) Forums dont require mentor to post guides, the wiki is free to edit. You don't need the role to be helpful, its to go even further beyond.

**Comment by PowerfulBacon** [comment]
For mentors seems off to me, considering the only thing required of them is to occasionally click on someones name and respond, people know that the expectation isnt insane and it's just there for experienced players to help new players without an insentive that can only really introduce a toxic environment

**Comment by mcmeiler** [comment]
> no? mentoring is literally un-abusable other than this. what are you gonna do, ban someone? there is literally nothing negative a mentor can do, only positive, and that's why a higher bar can only be a bad thing

There have been cases of mentors using meta-info from mhelps to give themselves an advantage.

**Comment by RDS88-beeman** [comment]
> 
> 
> > no? mentoring is literally un-abusable other than this. what are you gonna do, ban someone? there is literally nothing negative a mentor can do, only positive, and that's why a higher bar can only be a bad thing
> 
> There have been cases of mentors using meta-info from mhelps to give themselves an advantage.

and they disgraced the name of mentor and get sent to the shadow realm. They do not get the funny hats.

**Comment by ItsTisme** [comment]
uh based???? I get ian pack now

**Comment by francinum** [comment]
~~make giving it to admins and mentors a config~~

**Comment by Crossedfall** [comment]
> ~make giving it to admins and mentors a config~

Actually...... Not the worst idea.

**Comment by TheFakeElon** [comment]
M*ntors

**Comment by utilbird** [comment]
Mentors have a very small workload in comparison to the rest of the server staff and few if any additional responsibilities, my only concern is that people will apply _just_ to get donator items. Which could worsen if/when we expand donator perks. That's my 2 cents

**Comment by Crossedfall** [comment]
> Mentors have a very small workload in comparison to the rest of the server staff and few if any additional responsibilities, my only concern is that people will apply _just_ to get donator items. Which could worsen if/when we expand donator perks. That's my 2 cents

Up the requirements to compensate, this is all just a staffing and management issue more than an issue with the concept of rewarding mentors. 

We have some REALLY excellent mentors and I don't think they should be punished because we've historically had low standards. I'd much rather bring those standards up to their level and go from there.

**Comment by utilbird** [comment]
The culling :trollface:

**Comment by ItsTisme** [comment]
Maby having mentors and senior mentors, normal mentors work same way as is while senior mentors are closer to actuel staff to really teach newer players with extra perks/tools

**Comment by DatBoiTim** [comment]
> I support this for admins but idk about mentors.

This. While we could do for purging inactive mentors, not sure if there's enough responsibility as mentor to warrant access.

**Comment by DatBoiTim** [comment]
> Maby having mentors and senior mentors, normal mentors work same way as is while senior mentors are closer to actuel staff to really teach newer players with extra perks/tools

Would this be a functionality difference or just extra ranking?

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -2021,7 +2021,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
 		return
-	if(IS_PATRON(parent.ckey))
+	if(IS_PATRON(parent.ckey) || (parent in GLOB.admins) || (parent in GLOB.mentors))
```
Comment (Crossedfall) [inline code/modules/client/preferences.dm]:
```suggestion
	if(IS_PATRON(parent.ckey) || (parent in GLOB.admins))
```

Not in the headspace to actually add configs and deal with all that, so for now I'll just do admins and revisit mentors later.

**Comment by Crossedfall** [comment]
Currently reevaluating mentors as a whole, will come back to their access to donor items after that's all done and settled.

**Review by ike709 — APPROVED** [review]

**Comment by utilbird** [comment]
image

**Review by utilbird — APPROVED** [review]

**Pull request merged**